pyia/data.py

```diff
@@ -227,7 +227,11 @@ def pm(self):
     @property
     def distance(self):
         """Assumes 1/parallax. Has shape `(nrows,)`"""
-        return coord.Distance(parallax=self.parallax)
+        # Clip to positive values (1e-3 mas = 1 Mpc)
```

Comment:
I think I would prefer to default to `NaN` rather than a hard-coded clip value. But way to make us both happy could be to add a `get_distance()` method that looks something like this (I didn't test this):
```python
@u.quantity_input(fill_value=u.mas, clip_value=u.mas, 
                  equivalencies=u.parallax())
def get_distance(self, fill_value=None, clip_value=None):
    plx = self.parallax.copy()

    if clip_value is not None and fill_value is None:
        raise ValueError("If specifying a clip value, you must also")

    if clip_value is not None:
        plx[plx > clip_value.to(plx.unit, 
                                u.parallax())] = fill_value.to(plx.unit, 
                                                               u.parallax())

    if fill_value is not None:
        plx[plx <= 0] = fill_value.to(plx.unit, u.parallax())
        
    return coord.Distance(parallax=plx)

@property
def distance(self):
    """Assumes 1/parallax. Has shape `(nrows,)`. 

    This attribute will raise an error when  there are negative or zero parallax
    values. For more flexible retrieval of distance values and auto-filling bad
    values, use the .get_distance() method."""
    return self.get_distance()
```

Reply:
Oh, also, `get_distance()` would need a docstring.